src/reseeding.rs

```diff
@@ -51,12 +55,53 @@ impl<R: Rng, Rsdr: Reseeder<R>> ReseedingRng<R, Rsdr> {
 
     /// Reseed the internal RNG if the number of bytes that have been
     /// generated exceed the threshold.
+    /// 
+    /// On error, this may delay reseeding or not reseed at all.
     pub fn reseed_if_necessary(&mut self) {
         if self.bytes_generated >= self.generation_threshold {
-            self.reseeder.reseed(&mut self.rng);
+            loop {
+                if let Err(e) = self.reseeder.reseed(&mut self.rng) {
+                    // TODO: log?
+                    if e.kind.should_wait() {
+                        // Delay reseeding
+                        let delay = max(self.generation_threshold >> 8, self.bytes_generated);
+                        self.bytes_generated -= delay;
```

Comment:
Does it make sense to rename `bytes_generated` to something more descriptive of this delay? Maybe a countdown with `bytes_until_reseed`?

Reply:
It's count-up, not count-down. A more generic `bytes_count` would do, but it's less descriptive, so I'll leave as is.